plugin/trino-exasol/src/main/java/io/trino/plugin/exasol/ExasolClient.java

```diff
@@ -0,0 +1,290 @@
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding columns");
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming columns");
+    }
+
+    @Override
+    public void setColumnType(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column, Type type)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting column types");
+    }
+
+    @Override
+    public void dropColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping columns");
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, JdbcTableHandle handle)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping tables");
+    }
+
+    @Override
+    public void rollbackCreateTable(ConnectorSession session, JdbcOutputTableHandle handle)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables");
+    }
+
+    @Override
+    public String buildInsertSql(JdbcOutputTableHandle handle, List<WriteFunction> columnWriters)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support inserts");
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating schemas");
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName, boolean cascade)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping schemas");
+    }
+
+    @Override
+    public void renameSchema(ConnectorSession session, String schemaName, String newSchemaName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, Connection connection, String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
+    {
+        if (!remoteSchemaName.equals(newRemoteSchemaName)) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
+        }
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables");
+    }
+
+    @Override
+    protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCondition joinCondition)
+    {
+        // Deactivated because test 'testJoinPushdown()' requires write access which is not implemented for Exasol
```

Comment:
alternatively see `hasBehaviour` method in `BaseConnectorTest`. You can override it in Exasol to return `false` for `SUPPORTS_JOIN_PUSHDOWN` and then the test will be skipped.

Reply:
This would mean the connector reports to support join conditions but we skip tests.

`SUPPORTS_JOIN_PUSHDOWN` is already set to false.